test/files/jvm/scala-concurrent-tck.scala

```diff
@@ -90,6 +90,25 @@ trait FutureCallbacks extends TestBase {
     promise.success(-1)
   }
 
+  def stressTestNumberofCallbacks(): Unit = once {
```

Comment:
@PerformantData just added a JMH benchmark framework for some work on OpenHashMap (and it was just merged a couple days ago, https://github.com/scala/scala/pull/5061).  I'd see if you can hook into that, or do something in parallel.  For now, I'd recommend using JMH manually to test for regressions.  It takes forever to run, so you don't want to do it as part of CI or even a nightly build, but there's nothing more accurate for microbenchmarking these days.